src/core/proj/qgscoordinatetransform.cpp

```diff
@@ -666,14 +666,17 @@ QgsRectangle QgsCoordinateTransform::transformBoundingBox( const QgsRectangle &r
 
   // check if result bbox is geographic and is crossing 180/-180 line: ie. min X is before the 180° and max X is after the -180°
   bool doHandle180Crossover = false;
-  double xMin = std::fmod( x[0], 180.0 );
-  double xMax = std::fmod( x[nXPoints - 1], 180.0 );
-  if ( handle180Crossover
-       && ( ( direction == Qgis::TransformDirection::Forward && d->mDestCRS.isGeographic() ) ||
-            ( direction == Qgis::TransformDirection::Reverse && d->mSourceCRS.isGeographic() ) )
-       && xMin > 0.0 && xMin <= 180.0 && xMax < 0.0 && xMax >= -180.0 )
+  if ( nXPoints > 0 && x.size() >= nXPoints )
```

Comment:
shoud be enough (given that x.size() is nXPoints * nYPoints and both nXPoints and NYPoints are >= 1
```suggestion
  if ( nXPoints > 0 )
```

Reply:
Yes indeed. Thanks @rouault